node/podcontroller.go

```diff
@@ -342,11 +372,32 @@ func (pc *PodController) syncPodInProvider(ctx context.Context, pod *corev1.Pod)
 	// Check whether the pod has been marked for deletion.
 	// If it does, guarantee it is deleted in the provider and Kubernetes.
 	if pod.DeletionTimestamp != nil {
-		if err := pc.deletePod(ctx, pod.Namespace, pod.Name); err != nil {
+		obj, ok := pc.podForDeletion.Load(key)
+		if ok {
+			// We've tried to delete this pod before.
+			dPod := obj.(deleteablePod)
+			// If our termination grace period is equal to, or longer than the previous pod's termination grace period, we do not
+			// need to retry the delete.
+			if *dPod.pod.Spec.TerminationGracePeriodSeconds <= *pod.Spec.TerminationGracePeriodSeconds {
+				return nil
+			}
+		}
+
+		if err := pc.deletePod(ctx, pod); errdefs.IsNotFound(err) {
```

Comment:
Shouldn't we've a check here to do a Get always and do force removal if not found ?